openretina/utils/plotting.py

```diff
@@ -245,6 +248,79 @@ def polar_plot_of_direction_of_motion_responses(
     plt.gca().set_xticklabels([f"{x}°" for x in sorted_directions])
 
 
+class ColorMapper:
+    def __init__(self, cmap_name: str, vmin: float = 0, vmax: float = 1):
+        self.cmap_name = cmap_name
+        self.vmin = vmin
+        self.vmax = vmax
+        self.cmap = plt.get_cmap(self.cmap_name)
+        self.norm = plt.Normalize(vmin=self.vmin, vmax=self.vmax)
+
+    def get_color(self, number):
+        color = self.cmap(self.norm(number))
+        return color
+
+
+def plot_vector_field_resp_iso(x: np.ndarray, y: np.ndarray, gradient_dict: np.ndarray,
+                               resp_dict: np.ndarray, normalize_response: bool = False,
+                               rc_dict: dict[str, Any] = {},
+                               cmap: str = "hsv") -> None:
+    """
+    Plots a vector field response with isoresponse lines.
+
+    Parameters:
+    x (array-like): The x-coordinates of the grid points.
+    y (array-like): The y-coordinates of the grid points.
+    gradient_dict (ndarray): A dictionary containing response gradients at grid points.
+    resp_dict (ndarray): A dictionary containing responses at grid points.
+    normalize_response (bool, optional): If True, normalize the response data. Default is False.
+    colour_norm (bool, optional): If True, apply color normalization. Default is False.
+    rc_dict (dict, optional): A dictionary of rc settings to use in the plot. Default is an empty dictionary.
+    cmap (str, optional): The colormap to use for the plot. Default is "hsv".
+
+    Returns:
+    None
+    """
+
+    Z = resp_dict.transpose()
+    if normalize_response:
+        Z=Z/Z.max() * 100
+    gradient_grid = gradient_dict[:, 1:-1, 1:-1]
+    X, Y = np.meshgrid(x, x)
+
+    # Define levels for isoresponse lines
+    levels = np.linspace(Z.min(), Z.max(), 25)
+    # cm = ColorMapper("cool", vmin=gradient_norm_grid.min(),
+    #                  vmax=gradient_norm_grid.max())
+
+    with mpl.rc_context(rc_dict):
+        fig = plt.figure()
+
+        # Create a contour plot with isoresponse lines
+
+        plt.contourf(X, Y, Z, levels=levels, cmap=cmap, zorder=200)  # Change cmap to the desired colormap
+        cont_lines = plt.contour(X,Y,Z, levels=levels, cmap='jet_r',zorder=300)
+        plt.gca().clabel(cont_lines, inline=True, fmt='%1.0f',
+                         levels = cont_lines.levels[::2], colors="k", fontsize=5, zorder=400)
```

Comment:
```suggestion
                         levels = list(cont_lines.levels)[::2], colors="k", fontsize=5, zorder=400)
```
Apparently, cont_lines.levels is an Iterable, so something you can iterate over in a for loop (like `range(10)`), but that you cannot necessarily index. That why the type checks complain with:
```
openretina/utils/plotting.py:304: error: Value of type "Iterable[float]" is not indexable
```

To avoid that type error, you can convert that iterable to a list.